### DISCUSSION TITLE
Astro + Hono: Applying Tailwind to server's HTML

### DISCUSSION THREAD

**Discussion by Lippiece** [tailwindlabs/tailwindcss#13653] [Help] [answered]
Hi. I'm using Hono to partially render content in my app with the frontend on Astro with Tailwind integration. Some parts of the frontend are placeholders that query API and get HTML from there. That HTML contains Tailwind classes. However, they don't work if I don't have these classes somewhere in the frontend by obvious reasons (Tailwind doesn't have a clue about the server). 

So, is there a way to solve this without creating shim tags with classes that are on the server?

Thanks!

**Comment by wongjn** [comment, marked as answer]
[Safelisting](https://tailwindcss.com/docs/content-configuration#safelisting-classes) perhaps?

**Reply by Lippiece** [reply]
Interesting. Perhaps not the most convenient, but the solution that was built for my use case it seems.
